ent/encryption/replicated_key_provider.cc

```diff
@@ -347,6 +322,39 @@ future<std::tuple<UUID, key_ptr>> replicated_key_provider::get_key(const key_inf
     co_return std::tuple(uuid, k);
 }
 
+future<std::optional<std::tuple<UUID, key_ptr>>> replicated_key_provider::find_key(key_id& id, sstring_view cipher) {
```

Comment:
does not really need to return an optional since both uuid and key_ptr mustn't be null for a valid found key, no?

Reply:
It just adds complexity.
What should we do in case we return an engaged optional that contains null uuid and/or null key_ptr?
Since they all need to be engaged at the same time I don't see a point in wrapping the tuple in std::optional

Reply:
Actually, null UUID can be valid for other providers. This is an interface implementation.